pkgs/development/python-modules/pydrive2/default.nix

```diff
@@ -0,0 +1,28 @@
+{ lib, python3 }:
+
+python3.pkgs.buildPythonPackage rec {
```

Comment:
you shouldn't import a specific python version please look at https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#handling-dependencies

Reply:
Before trying to handle Python 2 vs 3, I received the error: `google-api-core-1.22.1 not supported for interpreter python2.7.` I'm reasonably confident this comes from...

https://github.com/NixOS/nixpkgs/blob/33402a812d5c0addb206d15e1426df007eea8b73/pkgs/development/python-modules/google_api_core/default.nix#L9

... transitively due to `google_api_python_client`.

I was unable to get incantations of `disabled = ! isPy3k` or `disabled = isPy27` to work... builds still gave me a Python 2 interpreter error. I ended up with a pattern that worked by cribbing from other PRs.

What am I missing?

Reply:
@jonringer ... thoughts?

Reply:
Using `pythonXPackages` isn't allowed from `python-modules` as this will mix interpreter versions for a given package. For example, if you were to do `python37Packages.<mypkg>`, but the expression pulled from `python3[8]Packages`. Python will be mis-importing several libraries, and this will likely cause issues at runtime.

Please refer to [Python dependency handling](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#handling-dependencies) on the best practices for handling dependencies within `python-modules`. Namely, you should be importing each python package individually, and not refer to another python package set (E.g. `python3Packages`).

If you're new to nixpkgs, you're free to view some of my videos on nixpkgs and python:
- [Python packaging using nixpkgs](https://www.youtube.com/watch?v=jXd-hkP4xnU)
- [Adding a package to nixpkgs unstable](https://www.youtube.com/watch?v=fvj8H5yUKu8)

Reply:
convert it over to using the individual packages and have it build for python3, and we'll go from there :)

Reply:
I picked this back up. I think I did all the needful in 5b86966d9a6cbb4d35419cd370d467727887bc40